docs/guide/writing-tasks.md

```diff
@@ -110,31 +110,37 @@ android_test_task:
 
 ## Cache Instruction
 
-A `cache` instruction allows to save some folder in cache based on a fingerprint and reuse it during the next execution 
-of the task with the same fingerprint. A `cache` instruction can be named the same way as `script` instruction.
+A `cache` instruction allows you to save a directory in the cloud. It will be downloaded onto the machine given to your task when new builds are triggered. This can help save time for installing dependencies, or just storing a large folder. A `cache` instruction can be named the same way as `script` instruction.
 
 Here is an example:
 
 ```yaml
 test_task:
+  container:
+    image: node:latest
   node_modules_cache:
     folder: node_modules
     fingerprint_script: cat yarn.lock
     populate_script: yarn install
   test_script: yarn run test
 ```
 
-A `fingerprint_script` is an optional field that can specify a script that will be executed and console output of which
-will be used as a fingerprint for the given task. By default task name is used as a fingerprint value.
+The `folder` is a *required* field that tells the agent which folder to cache. It should be relative to the working directory, or the root directory of the machine (ex. `node_modules` or `/usr/bin/bundler`).
+
+A `fingerprint_script` is an *optional* field that can specify a script that will be executed and console output of which
+will be used as a be used as a key for the given cache. By default the task name is used as a fingerprint value.
 
-`populate_script` is an optional field that can specify a script that will be executed to populate the cache. 
-`populate_script` should create `folder` if it doesn't exist before the `cache` instruction.
+`populate_script` is an *optional* field that can specify a script that will be executed to populate the cache.
+`populate_script` should create the `folder` if it doesn't exist before the `cache` instruction.
+If your dependencies are updated often, we suggest making the `fingerprint` script log something different so new versions of dependencies are used.
```

Comment:
Maybe something like:

```suggestion
If your dependencies are updated often, please pay attention to `fingerprint_script` and make sure it will produce different outputs for different versions of your dependency (ideally just print locked versions of dependencies).
```